application/actions/upload.go

```diff
@@ -0,0 +1,59 @@
+package actions
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/silinternational/wecarry-api/domain"
+
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/render"
+	"github.com/silinternational/wecarry-api/models"
+)
+
+// FileFieldName is the multipart field name for the file upload.
+const FileFieldName = "file"
+
+// UploadResponse is a JSON response for the /upload endpoint
+type UploadResponse struct {
+	Error       *domain.AppError `json:"Error,omitempty"`
+	Name        string           `json:"filename,omitempty"`
+	UUID        string           `json:"id,omitempty"`
+	URL         string           `json:"url,omitempty"`
+	ContentType string           `json:"content_type,omitempty"`
+	Size        int              `json:"size,omitempty"`
+}
+
+// UploadHandler responds to POST requests at /upload
+func UploadHandler(c buffalo.Context) error {
+	f, err := c.File(FileFieldName)
+	if err != nil {
+		return c.Render(http.StatusInternalServerError, render.JSON(UploadResponse{
```

Comment:
Maybe we should include a TODO comment about logging these errors by calling a function that calls domain.Error(), similar to actions.authRequestError()